test/lib/mock-blpapi.ts

```diff
@@ -0,0 +1,279 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+import assert = require('assert');
+import events = require('events');
+import util = require('util');
+import Promise = require('bluebird');
+import _ = require('lodash');
+import blpapi = require('blpapi');
+
+// Re-export blpapi public interfaces
+/* tslint:disable:interface-name */
+export interface SessionOpts extends blpapi.SessionOpts {}
+export interface Subscription extends blpapi.Subscription {}
+export interface Identity extends blpapi.Identity {}
+/* tslint:enable:interface-name */
+export interface ISession extends blpapi.ISession {}
+
+export interface IInstruction {
+    start?: boolean;
+    openService?: boolean;
+    stop?: boolean;
+    request?: string[];
+
+    [index: string]: any;
+}
+
+// ipc channel through websocket
+export var ipc: SocketIO.Socket;
+export var instructions: IInstruction = {};
+
+// Mapping of request types to response names to listen for.
+// The strings are taken from section A of the BLPAPI Developer's Guide, and are organized by
+// service.
+var REQUEST_TO_RESPONSE_MAP: { [index: string]: string; } = {
```

Comment:
These maps are duplicated with the wrapper (and are out of date since auth got merged after this change was started). Eventually they should be shared, although I'm ok if that happens after we move the wrapper into the node project if that's easier for code organization reasons. If we wait, add a TODO comment.